misc/python/materialize/mzbuild.py

```diff
@@ -1009,10 +1011,22 @@ def acquire(self, max_retries: int | None = None) -> None:
 
         # Only retry in CI runs since we struggle with flaky docker pulls there
         if not max_retries:
-            max_retries = 3 if ui.env_is_truthy("CI") else 1
+            max_retries = 5 if ui.env_is_truthy("CI") else 1
         assert max_retries > 0
 
-        deps_to_build = [dep for dep in self if not dep.try_pull(max_retries)]
+        deps_to_build = [
+            dep for dep in self if not dep.publish or dep.try_pull(max_retries)
```

Comment:
No need to try to pull an image that we never publish, becomes slower with this PR